apps/angular/signal-input/src/app/user.component.ts

```diff
@@ -19,23 +19,19 @@ const ageToCategory = (age: number): Category => {
   standalone: true,
   imports: [TitleCasePipe],
   template: `
-    {{ fullName | titlecase }} plays tennis in the {{ category }} category!!
+    {{ $fullName() | titlecase }} plays tennis in the
+    {{ $category() }} category!!
   `,
   host: {
     class: 'text-xl text-green-800',
   },
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class UserComponent implements OnChanges {
-  @Input({ required: true }) name!: string;
-  @Input() lastName?: string;
-  @Input() age?: string;
+export class UserComponent {
+  $name = input.required<string>({ alias: 'name' });
+  $lastName = input('', { alias: 'lastName' });
+  $age = input(0, { alias: 'age', transform: Number });
 
-  fullName = '';
-  category: Category = 'Junior';
-
-  ngOnChanges(): void {
-    this.fullName = `${this.name} ${this.lastName ?? ''}`;
-    this.category = ageToCategory(Number(this.age) ?? 0);
-  }
+  $fullName = computed(() => `${this.$name()} ${this.$lastName()}`);
+  $category = computed(() => ageToCategory(Number(this.$age())));
```

Comment:
why do you use Number again here, `age` should already be of type number

Reply:
you right, missed it